### PR TITLE
[REFACTOR] Clearly distinguish prod and development css

### PR DESCRIPTION
**Pull request by rix0rrr**
After the discussion in #4102 and #4130, I will try to clean up the mess I made!

Here is the new situation:

* We now have two different scripts, `generate-prod-css` and `generate-development-css`.
* These generate into two different files, `generated.full.css` and `generated.css`. The `generated.css` file is `gitignore`d.
* Depending on whether we are in debug mode or not, the right file is referenced in the HTML.

Hopefully, this will get rid of all the confusion surrounding the purpose of these files!

@TiBiBa, @reedspool, what do you think?

### CODE REVIEW

**Comment by rix0rrr** [comment]
Here's another thought: we can potentially NOT minify the files that get committed to version control, to reduce the chances of merge conflicts in them!

**Comment by TiBiBa** [comment]
> Here's another thought: we can potentially NOT minify the files that get committed to version control, to reduce the chances of merge conflicts in them!

Would love this! Not minifying the development version makes it easier to understand for new contributors as well, with the addition that CSS changes in a PR can be clearly distinguished in the file.

**Review by Felienne — CHANGES_REQUESTED** [review]
Setting this to changes requires for the not minifying (otherwise it is too hard for me to keep track of the PRs)

**Comment by reedspool** [comment]
Nice! Could consider `.gitignore` for the deployment CSS now.

This definitely feels like a great step especially and much more informative docs!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -204,29 +204,47 @@ Make sure to reload your browser (and work in incognito mode) to see the changes
...
+library](https://tailwindcss.com/docs/installation).  This library has generated
+classes for styling which we can apply to HTML elements.
+
+You normally do not need to think about this. During development, a multi-megabyte
+CSS file will be served that contains most classes. During deployment, a minimized CSS
+file is automatically produced.
+
+You may need to regenerate the development CSS file if you want to do one of the following things:
+
+* Want to use a *conditional* Tailwind class (for example, a class that starts with `hover:`).
```
Comment (rix0rrr) [inline CONTRIBUTING.md:218]:
```suggestion
* Use a *conditional* Tailwind class (for example, a class that starts with `hover:`).
```

**Comment by Felienne** [comment]
> LGTM!

Do you have enough rights to officially approve the PR? You have my blessing to do so!

**Comment by reedspool** [comment]
I do not see a button to do so

**Pull request merged**